### DISCUSSION TITLE
posts/remote-r/

### DISCUSSION THREAD

**Discussion by giscus** [mvuorre/mvuorre.github.io#22] [giscus]
# posts/remote-r/

Running R on a remote computer is surprisingly easy

https://vuorre.netlify.app/posts/remote-r/

**Comment by mvuorre** [comment]
Just a comment before I get some time to edit the source: DO MOUNT a volume to the docker container so your saved work isn't wiped if you have to restart the container. 

Courtesy of the OG 🧙 [Kristoffer](https://mastodon.rpsychologist.com/@kristoffer/109450908246501453)

**Comment by meflynn** [comment]
Finally got to get this setup today. Thanks for putting it together! I think it's working correctly, but I'm getting hung up on the last step where I connect. I might need to go back to the office computer and check the firewall settings.

**Reply by mvuorre** [reply]
💎 

I'd make sure that you've made tailscale run always. It's also possible that your workplace network has some setup that prevents these kinds of VPNs. In which case 😢. Good luck!

**Reply by meflynn** [reply]
Will do! I looked at the admin console and it looks like both are connected, and the browser message seems to be a "timed out" sort of warning, but I'll double check tomorrow. If this works it will be a huge relief. I really dreaded having to use some terminal based ssh sort of language. I did it with the university cluster, and it's not the worst, but it's also not the best.

**Comment by nxskok** [comment]
thank you for the nice clear writeup! (I have a desktop and a laptop here at home, so I can try this now.)

**Reply by mvuorre** [reply]
Cheers, let me know if there's any hiccups--I think this should be quite robust.

**Comment by JSLJ23** [comment]
The docker image runs but when I log in to the R studio page it shows:
`Could not connect to the R session on RStudio Server.`
`Unable to connect to service (1)`

And the terminal shows these errors:
```
023-07-04T16:34:54.174023Z [rsession-rstudio] ERROR system error 13 (Permission denied) [path: /home/rstudio/.local/share/rstudio, target-dir: ]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::createDirectory(const string&) const src/cpp/shared_core/FilePath.cpp:846; LOGGED FROM: int main(int, char* const*) src/cpp/session/SessionMain.cpp:2214
2023-07-04T16:35:03.991561Z [rserver] ERROR RStudio session start failed for: rstudio: in 10.2s error: rpc error: system error 2 (No such file or directory) [host: xxx.xxx.xxx.xxx:8787, uri: /rpc/client_init, path: /var/run/rstudio-server/rstudio-rsession/rstudio-d, user-id: 1000]; OCCURRED AT void rstudio::core::http::LocalStreamAsyncClient::handleConnect(const rstudio_boost::system::error_code&) src/cpp/server/session/ServerSessionProxy.cpp:124; LOGGED FROM: void rstudio::server::SessionManager::removePendingLaunch(const rstudio::core::r_util::SessionContext&, bool, const string&) src/cpp/server/session/ServerSessionManager.cpp:423
```

The ip address has been changed to xxx.
